modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -3114,7 +3114,12 @@ def _groupby_internal_columns(self, by, drop):
         else:
             if not isinstance(by, list):
                 by = [by] if by is not None else []
-            internal_by = [o for o in by if hashable(o) and o in self.columns]
+            internal_by = []
+            for o in by:
+                if isinstance(o, pandas.Grouper) and o.key in self.columns:
+                    internal_by.append(o.key)
```

Comment:
Will this work if the `pandas.Grouper` object's `freq is not None` ?

Reply:
@devin-petersohn yeah, we have tests that are already testing for that